src/v/kafka/server/server.cc

```diff
@@ -515,21 +515,27 @@ ss::future<response_ptr> list_groups_handler::handle(
     resp.data.error_code = error;
     resp.data.groups = std::move(groups);
 
-    if (ctx.authorized(
-          security::acl_operation::describe, security::default_cluster_name)) {
-        co_return co_await ctx.respond(std::move(resp));
-    }
+    auto cluster_authz = ctx.authorized(
+      security::acl_operation::describe, security::default_cluster_name);
+
+    if (!cluster_authz) {
+        // remove groups from response that should not be visible
+        auto non_visible_it = std::partition(
+          resp.data.groups.begin(),
+          resp.data.groups.end(),
+          [&ctx](const listed_group& group) {
+              return ctx.authorized(
+                security::acl_operation::describe, group.group_id);
+          });
 
-    // remove groups from response that should not be visible
-    auto non_visible_it = std::partition(
-      resp.data.groups.begin(),
-      resp.data.groups.end(),
-      [&ctx](const listed_group& group) {
-          return ctx.authorized(
-            security::acl_operation::describe, group.group_id);
-      });
+        resp.data.groups.erase(non_visible_it, resp.data.groups.end());
+    }
 
-    resp.data.groups.erase(non_visible_it, resp.data.groups.end());
+    if (!ctx.audit()) {
+        resp.data.groups.clear();
+        resp.data.error_code = error_code::broker_not_available;
+        co_return co_await ctx.respond(std::move(resp));
+    }
```

Comment:
This needs an additional resource callback to lsit groups in case cluster_authz passes